vendor/github.com/coreos/etcd/raft/raft.go

```diff
@@ -525,6 +580,20 @@ func (r *raft) becomeCandidate() {
 	r.logger.Infof("%x became candidate at term %d", r.id, r.Term)
 }
 
+func (r *raft) becomePreCandidate() {
+	// TODO(xiangli) remove the panic when the raft implementation is stable
```

Comment:
This looks mildly concerning.

@xiang90: Is this referring to any specific instability, or is it just a note to remove the panic eventually?